src/Evaluator.hs

```diff
@@ -311,20 +311,15 @@ doTypeCase blameLoc (Ty v0) ((p, rhs0) : ps) = match (doTypeCase blameLoc (Ty v0
     match :: Eval Value -> TypePattern -> Core -> TyF Ty -> Eval Value
     match next (TypePattern t) rhs scrut =
       case (t, scrut) of
-        (TSyntax, TSyntax) -> eval rhs
-        (TSignal, TSignal) -> eval rhs
```

Comment:
`(TString, TString)` was missing, so previously a type-case on `String` would mysteriously fail to match:

```
-- Internal error during expansion! This is a bug in the implementation.
-- Pattern info not added
(define-macro (m)
  (>>= (which-problem)
       (lambda (prob)
         (case prob
           [(expression t)
            (type-case t
              [String
               (pure '"foo")])]))))

(example (the String (m)))
```

With this PR it, all the type constructors are treated generically, so we can no longer forget to add a case... but a type-case on `String` still fails with the same error message. I am thus guessing that the error occurs earlier.

Reply:
I suspect without having looked that the `String` builtin doesn't expand to the right thing in a type-pattern context, if this is what you're seeing. In any case, making it generic is definitely an improvement!